### PR TITLE
Submission review service support

### PR DESCRIPTION
**Pull request by ekkinox**
feat: added [submission review service](https://www.imsglobal.org/spec/lti-sr/v1p0) support (line item submission review)

target release: `1.2.0`

### CODE REVIEW

File: src/Factory/LineItem/LineItemSubmissionReviewFactory.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+class LineItemSubmissionReviewFactory implements LineItemSubmissionReviewFactoryInterface
+{
+    /**
+     * @throws InvalidArgumentException
+     */
+    public function create(array $data): LineItemSubmissionReviewInterface
+    {
+        if (!array_key_exists('reviewableStatus', $data)) {
+            throw new InvalidArgumentException('Missing mandatory reviewableStatus');
```
Comment (wazelin) [inline src/Factory/LineItem/LineItemSubmissionReviewFactory.php:37]: Sounds more like a `DomainException` to me.
`InvalidArgumentException`'s purpose is to indicate the argument has an unexpected _type_.

File: src/Model/LineItem/LineItemSubmissionReview.php [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        return array_filter(
+            [
+                'reviewableStatus' => $this->reviewableStatuses,
+                'label' => $this->label,
+                'url' => $this->url,
+                'custom' => $this->customProperties,
+            ],
+            static function ($value): bool {
+                return !empty($value);
+            }
```
Comment (wazelin) [inline src/Model/LineItem/LineItemSubmissionReview.php:127]: Isn't that condition basically a default behavior of `array_filter()`?
```suggestion
            ]
```

**Review by wazelin — CHANGES_REQUESTED** [review]

File: src/Factory/LineItem/LineItemSubmissionReviewFactory.php (hunk 1 above)
Comment (ekkinox) [reply src/Factory/LineItem/LineItemSubmissionReviewFactory.php:37]: Every input checks in LTI framework, when invalid, lead to InvalidArgumentException, to lead when it bubbles up to a bad request http exception.

I get your point on domain exception usage, and I can agree, but if we apply domain usage here, we should refactor everywhere.

File: src/Model/LineItem/LineItemSubmissionReview.php (hunk 2 above)
Comment (ekkinox) [reply src/Model/LineItem/LineItemSubmissionReview.php]: true, removed, thx

**Review by wazelin — APPROVED** [review]
LGTM, thanks a bunch!

File: src/Factory/LineItem/LineItemSubmissionReviewFactory.php (hunk 1 above)
Comment (ekkinox) [reply src/Factory/LineItem/LineItemSubmissionReviewFactory.php:37]: InvalidArgumentException and DomainException are btw both extending LogicException (that should lead directly to a fix in the code) => makes imo not better sense.

bad user input should not lead to us changing our code, if one day we refine the exception for those situations, I would actually go more in the direction of RuntimeException usage, wdyt @wazelin  ?

**Review by peetya — APPROVED** [review]

File: src/Factory/LineItem/LineItemSubmissionReviewFactory.php (hunk 1 above)
Comment (wazelin) [reply src/Factory/LineItem/LineItemSubmissionReviewFactory.php:37]: Yeah, that's making a lot more sense.

**Pull request merged**